testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/servlet/methods/DenyUncoveredHttpMethodsTestCase.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * @author Jan Tymel
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(WebTestsSecurityDomainSetup.class)
+public class DenyUncoveredHttpMethodsTestCase {
+
+    @ArquillianResource(SecuredServlet.class)
+    URL deploymentURL;
+
+    private static final String CONTENT_FROM_GET = "GET";
+    private static final String CONTENT_FROM_HEAD = "HEAD";
+
+    @Test
+    public void testCorrectUserAndPassword() throws Exception {
+        HttpGet httpGet = new HttpGet(getURL());
+        HttpResponse response = getHttpResponse(httpGet, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_OK));
+
+        assertThat(contentOf(response), is(CONTENT_FROM_GET));
+    }
+
+    @Test
+    public void testHeadMethod() throws Exception {
+        HttpHead httpHead = new HttpHead(getURL());
+        HttpResponse response = getHttpResponse(httpHead, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_OK));
+
+        assertThat(testHeaderContentOf(response), containsString(CONTENT_FROM_HEAD));
+    }
+
+    @Test
+    public void testPostMethod() throws Exception {
+        HttpPost httpPost = new HttpPost(getURL());
+        HttpResponse response = getHttpResponse(httpPost, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_FORBIDDEN));
+    }
+
+    @Test
+    public void testPutMethod() throws Exception {
+        HttpPut httpPut = new HttpPut(getURL());
+        HttpResponse response = getHttpResponse(httpPut, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_FORBIDDEN));
+    }
+
+    @Test
+    public void testDeleteMethod() throws Exception {
+        HttpDelete httpDelete = new HttpDelete(getURL());
+        HttpResponse response = getHttpResponse(httpDelete, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_FORBIDDEN));
+    }
+
+    @Test
+    public void testTraceMethod() throws Exception {
+        HttpTrace httpTrace = new HttpTrace(getURL());
+        HttpResponse response = getHttpResponse(httpTrace, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_FORBIDDEN));
+    }
+
+    @Test
+    public void testOptionsMethod() throws Exception {
+        HttpOptions httpOptions = new HttpOptions(getURL());
+        HttpResponse response = getHttpResponse(httpOptions, "anil", "anil");
+
+        assertThat(statusCodeOf(response), is(HttpServletResponse.SC_FORBIDDEN));
+    }
+
+    private HttpResponse getHttpResponse(HttpUriRequest request, String user, String password) throws IOException {
+        String auth = user + ":" + password;
+        byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(Charset.forName("US-ASCII")));
+        String authHeader = "Basic " + new String(encodedAuth);
+        request.setHeader(HttpHeaders.AUTHORIZATION, authHeader);
```

Comment:
Good point, thanks. Correct behavior can be tested without auth. Comparison of returned http status codes should be enough.